h/api/nipsa/search.py

```diff
@@ -0,0 +1,83 @@
+import copy
+
+
+def nipsa_filter(query, userid=None):
```

Comment:
That sounds like the kind of thing we may need in the future when we have a function for nipsa filtering, a function for group filtering, and a function for combining them.

But for now all logic about Es filters (including how to nest them in a query with a "filtered" dict containing "filter" and "query" keys) is encapsulated in this one function, which is nice.  I believe there's no code anywhere else that knows anything about Es filters.